podium/datasets/iterator.py

```diff
@@ -494,9 +493,12 @@ def __init__(self, dataset: DatasetBase = None, shuffle=True, add_padding=True):
             iterator. If set to ``False``, numericalized Fields will be
             returned as python lists of ``matrix_class`` instances.
         """
+
+        batch_size = len(dataset) if dataset else 0
```

Comment:
batch_size of 0 looks really weird? What exactly does it mean and how exactly does that fit in `range`

Reply:
Yeah, wasn't sure what to put here tbh. I didn't want to make the change we discussed in #269 where the `dataset` arg would be required because I'd also need to change a lot of dependent code. The iterator will error out during iteration if the dataset isn't set in any case, and the batch size will be set when the dataset is set so I think this is OK although not pretty.

Reply:
perhaps just raise an exception or set it to None?

Reply:
Set it to `None`, tests work fine. This will need to be addressed soonish though.